### PR TITLE
resolve #4085: work around bogus unregistration of document listeners on

### LINKED ISSUE #4085: `didOpen` events are missing for newly opened documents after reconnection
We've switched from multiple language clients to one per a language to avoid leaking them: https://github.com/theia-ide/theia/pull/3538#issuecomment-442889603

It's turned out that on `stop` language client disposes document listeners but not unset them, and on following `start` it does not install new listeners, so all document events are missing. And there is no language support at all.

In order to reproduce:
- restart ts language server
- open ts file
- now language support, but should be

### PR DESCRIPTION
**Pull request by akosyakov**
See https://github.com/theia-ide/theia/issues/4085 for details.

### CODE REVIEW

File: packages/languages/src/browser/language-client-factory.ts [hunk 1]
```diff
@@ -74,7 +74,30 @@ export class LanguageClientFactory {
                     return createConnection(connection, errorHandler, closeHandler);
                 }
             }
-        });
+        }));
+    }
+
+    /**
+     * see https://github.com/theia-ide/theia/issues/4085
+     * remove then monaco-languageclient is upgraded to latest vscode-languageclient
```
Comment (AlexTugarev) [inline packages/languages/src/browser/language-client-factory.ts]: 'when'

Comment (akosyakov) [reply packages/languages/src/browser/language-client-factory.ts]: starting a workspace just to fix it 🙄

Comment (AlexTugarev) [reply packages/languages/src/browser/language-client-factory.ts]: no! was just kidding

Comment (akosyakov) [reply packages/languages/src/browser/language-client-factory.ts]: too late, they are cheap and fast 😄

**Pull request merged**